### DISCUSSION TITLE
Unable to alter green and blue channels of this new image

### DISCUSSION THREAD

**Discussion by entvidya** [ImageMagick/ImageMagick#7184] [Help] [answered]
### ImageMagick version

7.1.1-29

### Operating system, version and so on

Windows 11

### Description

I am running the following command to achieve these goals:
- Copy the alpha channel from existing image1 into the red channel of new image2
- Set the green and blue channels of new image2 to black
- Copy the color (it's a greyscale image) of existing image3 into the alpha channel of new image2

End result being that image2 is a new file that contains a red channel that is a copy of image1's alpha channel, green and blue channels that are black, and an alpha channel that is a copy of any of image3's color channels.

When I run these commands, the output file that I receive has the red and alpha channels that I intended, but the blue and green channels are still a copy of image1's alpha channel, rather than being black. Is this a bug, or could somebody point out what I am doing wrong?

`magick image1.png -alpha extract image2.png`
`magick image2.png -channel G,B -fill black +channel image3.png -resize 1024x1024! -alpha off -compose copy-opacity -composite image2.png`

A copy of the images I was working on can be found here: https://drive.google.com/drive/folders/18OlhBufnGRVs0GTUCphWmAOOOQpa6GTf?usp=sharing

**Comment by snibgo** [comment, marked as answer]
`-fill Black` sets the colour to use for certain operations. By itself, it doesn't change the image. See https://imagemagick.org/script/command-line-options.php#fill

Perhaps you intended `-fill Black -colorize 100`.

**Reply by entvidya** [reply]
This did exactly what I wanted, thank you.
